### PR TITLE
CLOUD-591 feature: Add support for sidecar containers for operators

### PR DESCRIPTION
**Pull request by ollevche**
[CLOUD-591](https://jira.percona.com/browse/CLOUD-591) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/CLOUD-591

Added sidecar containers, rerun types code generation and extended e2e tests

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | failed |
| users | failed |
| upgrade | passed |
| upgrade-sharded | passed |

File: e2e-tests/init-deploy/conf/another-name-rs0.yml [hunk 1]
```diff
@@ -59,6 +59,19 @@ spec:
...
             storage: 1Gi
+    sidecars:
+    - image: busybox
+      command: ["/bin/sh"]
+      args: ["-c", "while true; do echo echo $(date -u) 'test' >> /dev/null; sleep 5;done"]
+      name: my-sidecar-1
+    - image: busybox
+      command: ["/bin/sh"]
+      args: ["-c", "while true; do echo echo $(date -u) 'test' >> /dev/null; sleep 5;done"]
+      name: my-sidecar-1
```
Comment (delgod) [inline e2e-tests/init-deploy/conf/another-name-rs0.yml:70]: @spron-in what operator should do (print error/warning or ignore) if the user has specified 1) container with the name which used for operator containers 2) two containers with the same name?

**Review by delgod — CHANGES_REQUESTED** [review]

File: e2e-tests/init-deploy/conf/another-name-rs0.yml (hunk 1 above)
Comment (spron-in) [reply e2e-tests/init-deploy/conf/another-name-rs0.yml:70]: We need to inherit the behavior from Kubernetes for this case. This feature is for experienced users and additional failsafe mechanisms will not be implemented.

Comment (ollevche) [reply e2e-tests/init-deploy/conf/another-name-rs0.yml:70]: Let me summarize the behavior:

1. Operator prints an error and skips those containers

2. Operator does nothing and lets kubernetes handle those containers

@spron-in is it right?

Comment (spron-in) [reply e2e-tests/init-deploy/conf/another-name-rs0.yml:70]: It is correct

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| scheduled-backup | failed |
| security-context | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| self-healing | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| demand-backup-sharded | failed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| scheduled-backup | failed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| limits | passed |
| self-healing | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| scheduled-backup | failed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup | failed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-533-041b0d3d

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| users | passed |
| upgrade | passed |
| upgrade-sharded | passed |

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**